apps/demos-e2e/src/plugins/index.js

```diff
@@ -1,22 +1,154 @@
+  'disabled-by-default-v8.cpu_profiler',
+  'disabled-by-default-v8.cpu_profiler.hires',
+];
 
 module.exports = (on, config) => {
   // `on` is used to hook into various events Cypress emits
   // `config` is the resolved Cypress config
 
   // Preprocess Typescript file using Nx helper
   on('file:preprocessor', preprocessTypescript(config));
+
+  on('before:browser:launch', (browser, launchOptionsOrArgs) => {
+    debug('browser launch args or options %o', launchOptionsOrArgs);
+    const args = Array.isArray(launchOptionsOrArgs)
+      ? launchOptionsOrArgs
+      : launchOptionsOrArgs.args;
+
+    port = ensureRemoteDebuggingPort(args);
+    debug('ensureRemoteDebuggingPort %d', port);
+    debug('Chrome arguments %o', args);
+  });
+
+  on('task', {
+    resetCRI: async () => {
+      if (client) {
+        debug('resetting CRI client');
+        await client.close();
+        client = null;
+      }
+
+      return Promise.resolve(true);
+    },
+
+    enableProfiler: async () => {
+      debug('enableProfiler');
+      client = client || (await CDP({ port }));
+
+      const { Profiler } = client;
+      await Profiler.setSamplingInterval({ interval: 100 });
+
+      return Profiler.enable();
+    },
+
+    disableProfiler: async () => {
+      debug('disableProfiler');
+      client = client || (await CDP({ port }));
+
+      return client.Profiler.disable();
+    },
+
+    startProfiler: async () => {
+      debug('startProfiler');
+      client = client || (await CDP({ port }));
+
+      return client.Profiler.start();
+    },
+
+    stopProfiler: async (params) => {
+      debug('stopProfiler');
+      client = client || (await CDP({ port }));
+
+      const profilerResults = await client.Profiler.stop();
+
+      fs.writeFileSync(
+        `./profiling${params.suffix ? '_' + params.suffix : ''}_${currentDateString()}.cpuprofile`,
+        JSON.stringify(profilerResults.profile || 'Profile not found')
+      );
+
+      return Promise.resolve(profilerResults);
+    },
+
+    startTracing: async () => {
+      debug('startTracing');
+      client = client || (await CDP({ port }));
+      const { Page, Tracing } = client;
+
+      await Page.enable();
+      tracingEvents = [];
+      Tracing.dataCollected(({ value }) => {
+        tracingEvents.push(...value);
+      });
+
+      return Tracing.start({
+        categories: TRACE_CATEGORIES.join(','),
+        options: 'sampling-frequency=10000',
+      });
+    },
+
+    endTracing: async (params) => {
+      debug('endTracing');
+      client = client || (await CDP({ port }));
+
+      await client.Tracing.end();
+      const tracingComplete = await client.Tracing.tracingComplete();
+
+      fs.writeFileSync(
+        `./tracing${params.suffix ? '_' + params.suffix : ''}_${currentDateString()}.json`,
+        JSON.stringify(tracingEvents)
+      );
+
+      return Promise.resolve(tracingComplete);
+    },
+  });
 };
+
+function ensureRemoteDebuggingPort(args) {
```

Comment:
Maybe tiny comments why this code is here etc?

Reply:
Sure, I've added some more to be more clear for any future devs!